mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -498,8 +373,106 @@ impl Nexus {
         Ok(())
     }
 
+    /// generate a new nexus label based on the nexus configuration. The meta
+    /// partition is fixed in size and aligned to a 1MB boundary
+    pub(crate) fn generate_label(&mut self) -> NexusLabel {
+        let mut hdr = GPTHeader::new(
+            self.bdev.block_size(),
+            self.min_num_blocks(),
+            self.bdev.uuid().into(),
+        );
+
+        let mut entries = vec![GptEntry::default(); hdr.num_entries as usize];
+
+        entries[0] = GptEntry {
+            ent_type: GptGuid::from_str("27663382-e5e6-11e9-81b4-ca5ca5ca5ca5")
+                .unwrap(),
+            ent_guid: GptGuid::new_random(),
+            // 1MB aligned
+            ent_start: hdr.lba_start,
+            // 4MB
+            ent_end: hdr.lba_start
+                + ((4 << 20) / self.bdev.block_size()) as u64
+                - 1,
+            ent_attr: 0,
+            ent_name: GptName {
+                name: "MayaMeta".into(),
+            },
+        };
+
+        entries[1] = GptEntry {
+            ent_type: GptGuid::from_str("27663382-e5e6-11e9-81b4-ca5ca5ca5ca5")
+                .unwrap(),
+            ent_guid: GptGuid::new_random(),
+            ent_start: entries[0].ent_end + 1,
+            ent_end: hdr.lba_end,
+            ent_attr: 0,
+            ent_name: GptName {
+                name: "MayaData".into(),
+            },
+        };
+
+        // entries.resize(128, GptEntry::default());
```

Comment:
is this comment left alone as a remark for future task or is it stale?

Reply:
Should be removed, thanks!